bycycle/spikes/objs.py

```diff
@@ -240,21 +217,21 @@ def generate_spikes(self):
         """Generate spikes from fit parameters."""
 
         if self.df_features is None or self.params is None:
-            raise ValueError('The fit method must be successfully called prior to generating, '
-                             'using either n_gaussians = {2, 3}.')
+            raise ValueError('The fit method must be successfully called prior to generating ')
 
         self.spikes_gen = []
         for idx, param in enumerate(self.params):
-
+            print("idx", idx)
+            print("param1", param)
             if np.isnan(param[0]):
                 self.spikes_gen.append(np.nan)
                 continue
 
             times_spike = np.arange(0, len(self._spikes[idx])/self.fs, 1/self.fs)
 
             param = param[~np.isnan(param)]
-
-            spike_gen = sim_action_potential(times_spike, *param)
+            print("param",param)
+            spike_gen = sim_gaussian_cycle(times_spike, *param)
```

Comment:
Yes the currents are now generated separately - I can write a function that gets all parameters to generate one spike. Would that go in neurodsp or in the spike class?

Reply:
There's a function in neurodsp that allows spike simulation given a variable number of gaussians (`neurodsp.sim.transients.sim_action_potential` in the main branch but not pypi released yet). This would require a reshaping of the `self.params` to be compatible.

Alternatively, each index in `self.spike_gen` could be a 2d array with shape == (3 gaussians, n_timepoints), where taking the sum of axis=0 returns the combined spike.